src/services/userService.ts

```diff
@@ -60,9 +62,11 @@ export async function getUser(
 ): Promise<{ user?: UserBasicInfo; error?: string }> {
     try {
         await connectDB();
-        // TODO: Get USER_ID from session
-        const userId = stringToObjectId(userID);
-
+        const session = await getServerSession(authOptions);
+        if (!session) {
+            return { error: 'User not found' };
+        }
+        const userId = stringToObjectId(session.user.id);
```

Comment:
Functions in this service have the userID in parameter. You may either 
1. remove it completely or 
2. use it by removing the default value `= '123456789012' `, letting it be optional and updating the assignment
```
export async function getUser(
    userID: string?
): Promise<{ user?: UserBasicInfo; error?: string }> {
    try {
        await connectDB();
        let userId: mongoose.Types.ObjectId | null;
        if (userID) {
            userId = stringToObjectId(userID);
        } else {
            const session = await getServerSession(authOptions);
            if (!session) {
                return { error: 'User not found' };
            }
            userId = stringToObjectId(session.user.id);
        }
```

Reply:
The benefit to keep the `userID` is that it allows us to have the flexibility to administrate user in the future. If we don't have plan to do so, we can remove it.

Reply:
Fixed by completely deleting it.